.travis.yml

```diff
@@ -134,6 +145,11 @@ install:
   - export ANDROID_NDK_r17="`pwd`/_ci/android-ndk-r17"
 
 script:
+  - | # build corrade-rc if cross compiling
+    if [ ! -z ${CROSS_COMPILE+x} ]; then
+      ./magnum-build-corrade-rc.sh
```

Comment:
Please try to use the `hunter_experimental_add_host_project` feature first, see https://github.com/ruslo/hunter/issues/1635#issuecomment-443826466

Reply:
Well, I cannot say I did not try 

See my commits here : https://github.com/pthom/hunter/commit/ff01c8840f40e29818d54b683bcfec33376ba74a

See the build log :
- Step 1 :  https://travis-ci.org/pthom/hunter/jobs/463004765#1184 
Here this is `hunter/examples/corrade/CMakeLists.txt` speaking
`````cmake
if(IOS OR ANDROID)
  # add cmake/host subdiretcory as host project to install protoc
  include(hunter_experimental_add_host_project)
  hunter_experimental_add_host_project(cmake/host)

  add_executable(Corrade::corrade-rc IMPORTED)
  set_property(TARGET Corrade::corrade-rc APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
  set_target_properties(Corrade::corrade-rc PROPERTIES IMPORTED_LOCATION_RELEASE "${HUNTER_HOST_ROOT}/bin/corrade-rc")

  set(CORRADE_RC_EXECUTABLE "${HUNTER_HOST_ROOT}/bin/corrade-rc")

  if(EXISTS ${CORRADE_RC_EXECUTABLE})
    message("Found CORRADE_RC_EXECUTABLE at ${CORRADE_RC_EXECUTABLE}")
  else()
    message("Did not find CORRADE_RC_EXECUTABLE at ${CORRADE_RC_EXECUTABLE}")
  endif()
  message(STATUS "Using imported corrade-rc from host: ${HUNTER_HOST_ROOT}/bin/corrade-rc")
endif(IOS OR ANDROID)

`````

- Step 2 :  https://travis-ci.org/pthom/hunter/jobs/463004765#1383

Here, it is `corrade/CMakeLists.txt` that speaks:

````cmake
if(CMAKE_CROSSCOMPILING)
    find_program(CORRADE_RC_EXECUTABLE corrade-rc)
    if(NOT CORRADE_RC_EXECUTABLE)
        message(FATAL_ERROR "Native `corrade-rc` executable, which is needed when crosscompiling, was not found")
    endif()
    add_executable(Corrade::rc IMPORTED)
    set_property(TARGET Corrade::rc PROPERTY IMPORTED_LOCATION ${CORRADE_RC_EXECUTABLE})
endif()
````

I'm out of ideas. Are you?

Reply:
> I'm out of ideas. Are you?

> https://github.com/pthom/hunter/commit/ff01c8840f40e29818d54b683bcfec33376ba74a

This code should be part of "corrade" sources, not in examples. User don't have to do anything special, everything should "just work" with `hunter_add_package(corrade)` even if you're cross-compiling.

See my comment about ICU package: https://github.com/ruslo/hunter/issues/1635#issuecomment-443826466

On user side everything still plain simple even for iOS/Android:
* https://github.com/ruslo/hunter/blob/v0.23.63/examples/ICU/CMakeLists.txt#L14-L17

Reply:
You can skip iOS/Android testing for now and deal with it later.